tests/test_antler_proj.py

```diff
@@ -0,0 +1,84 @@
+
+This script tests following keys:
+--create-project
+--add-app
+--add-lib
+--add-dep
+--remove-app
+--remove-lib
+--remove-dep
+--update-app
+--update-lib
+--update-dep
+--build-project
+--clean-build-project
+
+"""
+
+import glob
+import os
+import shutil
+import subprocess
+import pytest
+
+from common import DUNE_EXE, TEST_PATH
+
+
+TEST_PROJECT_DIR = TEST_PATH + "/" + "antler_test_dir"
+
+
+def remove_existing():
+    """ Remove an existing `TEST_PROJECT_DIR` dir. """
+
+    if os.path.exists(TEST_PROJECT_DIR):
+        print("Removing a test directory if exists: ", TEST_PROJECT_DIR)
+        shutil.rmtree(TEST_PROJECT_DIR)
+
+
+@pytest.mark.skip(reason="Skeeped until the release of antler-proj. "
+                         "See details in https://github.com/AntelopeIO/DUNE/issues/134")
+def test_antler():
+
+    remove_existing()
+
+    subprocess.run([DUNE_EXE, "--create-project", "test_proj", TEST_PROJECT_DIR], check=True)
+
+    # Add ------------
+
+    subprocess.run([DUNE_EXE, "--add-app", TEST_PROJECT_DIR, "test_app",  "c++",
+                    "compiler_opts", "linker_opts"], check=True)
+    res = subprocess.run([DUNE_EXE, "--debug", "--add-app", TEST_PROJECT_DIR, "test_app",  "c++",
+                          "compiler_opts", "linker_opts"], stdout=subprocess.PIPE)
+    assert b'already exists in project' in res.stdout
+
+    subprocess.run([DUNE_EXE, "--add-lib", TEST_PROJECT_DIR, "test_lib",  "c++",
+                    "compiler_opts", "linker_opts"], check=True)
+    res = subprocess.run([DUNE_EXE, "--debug", "--add-lib", TEST_PROJECT_DIR, "test_lib",  "c++",
+                          "compiler_opts", "linker_opts"], stdout=subprocess.PIPE)
+    assert b'already exists in project' in res.stdout
+
+    # this test is commented because adding of dependencies doesn't work
+    # subprocess.run([DUNE_EXE, "--add-dep", TEST_PROJECT_DIR, "test_app",  "test_dep",
+    #                 "larryk85/sample_contract", "0.0.1", "hash"], check=True)
+    # res = subprocess.run([DUNE_EXE, "--debug", "--add-dep", TEST_PROJECT_DIR, "test_app",  "test_dep",
+    #                       "larryk85/sample_contract", "0.0.1", "hash"],
+    #                      stdout=subprocess.PIPE)
+    # assert b'already exists for' in res.stdout
+
+
+# Remove ------------
+
+    subprocess.run([DUNE_EXE, "--remove-app", TEST_PROJECT_DIR, "test_app"], check=True)
+    res = subprocess.run([DUNE_EXE, "--debug", "--add-app", TEST_PROJECT_DIR, "test_app",  "c++",
+                          "compiler_opts", "linker_opts"], stdout=subprocess.PIPE)
+    assert b'already exists in project' not in res.stdout
+
+    subprocess.run([DUNE_EXE, "--remove-lib", TEST_PROJECT_DIR, "test_lib"], check=True)
+    res = subprocess.run([DUNE_EXE, "--debug", "--add-lib", TEST_PROJECT_DIR, "test_lib",  "c++",
+                          "compiler_opts", "linker_opts"], stdout=subprocess.PIPE)
+    assert b'already exists in project' not in res.stdout
+
```

Comment:
could you please add test for validate and build? This could be for this empty project as well.

Reply:
> could you please add test for validate and build? This could be for this empty project as well.

We agreed with @ScottBailey to postpone the tests development and skip this file from active tests because antler-proj in active development stage and the tests will get broken after each major update i.e. almost every day. I removed all broken tests from this file and skipped it from the active tests.